src/Feature/Checkout/code/Controllers/OrderController.cs

```diff
@@ -33,48 +36,48 @@ public OrderController(IOrderRepository orderRepository)
             this.orderRepository = orderRepository;
         }
 
-        [Route("get/{orderId}")]
-        public IHttpActionResult GetOrder(string orderId)
+        [ResponseType(typeof(CartModel))]
+        [HttpGet, ActionName("getOrder")]
+        public ActionResult GetOrder(string orderId)
         {
-            if (string.IsNullOrEmpty(orderId))
-            {
-                return this.JsonError($"{orderId} should be specified.", HttpStatusCode.BadRequest);
-            }
-
-            var model = this.orderRepository.GetOrderDetails(orderId);
-
-            if (model.Success)
-            {
-                return this.JsonOk(model.Data);
-            }
+            return this.Execute(() => this.orderRepository.GetOrderDetails(orderId));
+        }
 
-            return this.JsonError(model.Errors.ToArray(), HttpStatusCode.InternalServerError);
+        [ResponseType(typeof(OrderHistoryResultModel))]
```

Comment:
Return value is weird. If I'm asking for orders I should get back several (or one) order entities.
Investigate how it works on FE, and try to understand is it possible to return enumerable of entities.

Reply:
I have doubt that `ResponseType` can be used with MVC controller. Did you face any examples of such usage previously?